spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/metadata/JdbcMetadataStore.java

```diff
@@ -245,5 +253,10 @@ public String remove(String key) {
 		}
 		return null;
 	}
-
+	@Override
+	public Set<String> keySet() {
+		Set<String> keys = new HashSet<>();
+		keys.addAll(this.jdbcTemplate.queryForList(this.getAllKeysQuery, new Object[]{this.region}, String.class));
```

Comment:
There is this variant to simplify the code here: `queryForList(String sql, Class<T> elementType, @Nullable Object... args)`